lib/cisco_node_utils/tacacs_global.rb

```diff
@@ -33,9 +33,24 @@ def initialize(name)
       fail ArgumentError,
            "This provider only accepts an id of 'default'" \
            unless name.eql?('default')
+      enable unless TacacsGlobal.enabled
       @name = name
     end
 
+    # Check feature enablement
+    def self.enabled
+      config_get('tacacs_global', 'feature')
+    rescue Cisco::CliError => e
```

Comment:
This `rescue` should not be needed if you use the api from `feature.rb`  https://github.com/cisco/cisco-network-node-utils/blob/develop/lib/cisco_node_utils/feature.rb#L200

The getter will return `false` if the feature is not found.